tutorials/02_intermediate/gridspec.py

```diff
@@ -1,126 +1,196 @@
+fig2.tight_layout()
+
+#############################################################################
+# When you want to have plots of different sizes, however,
+# :mod:`~matplotlib.gridspec` becomes indispensable and provides a couple
+# of options.
+# The method shown here initializes a uniform grid specification,
+# and then uses typical numpy indexing and slices to allocate multiple
+# "cells" for a given subplot.
+
+fig3 = plt.figure()
+spec3 = gridspec.GridSpec(ncols=3, nrows=3)
+anno_opts = dict(xy=(0.5, 0.5), xycoords='axes fraction',
+                 va='center', ha='center')
+
+fig3.add_subplot(spec3[0, 0]).annotate('GridSpec[0, 0]', **anno_opts)
+fig3.add_subplot(spec3[0, 1:]).annotate('GridSpec[0, 1:]', **anno_opts)
+fig3.add_subplot(spec3[1:, 0]).annotate('GridSpec[1:, 0]', **anno_opts)
+fig3.add_subplot(spec3[1:, 1:]).annotate('GridSpec[1:, 1:]', **anno_opts)
+
+fig3.tight_layout()
+
+############################################################################
+# This example shows off using the ``width_ratios`` and ``height_ratios``
+# options.
+# These keyword arguments are lists of numbers.
+# Note that absolute values are meaningless, only their relative ratios
+# matter.
+# That means that ``width_ratios=[2, 4, 8]`` is equivalent to
+# ``width_ratios=[1, 2, 4]`` within equally wide figures.
+# We'll blindly create the axes within ``for`` loops since we won't need
+# them later.
+
+fig4 = plt.figure()
+widths = [2, 3, 1.5]
+heights = [1, 3, 2]
+spec4 = gridspec.GridSpec(ncols=3, nrows=3, width_ratios=widths,
+                          height_ratios=heights)
+for row in range(3):
+    for col in range(3):
+        ax = fig4.add_subplot(spec4[row, col])
+        label = 'Width: {}\nHeight: {}'.format(widths[col], heights[row])
+        ax.annotate(label, (0.1, 0.5), xycoords='axes fraction', va='center')
+
+fig4.tight_layout()
+
+############################################################################
+# Learning to use ``width_ratios`` and ``height_ratios`` is particularly
+# useful since :func:`~matplotlib.pyplot.subplots` accepts them within the
+# ``gridspec_kw`` parameter.
+# For that matter, any parameter accepted by
+# :class:`~matplotlib.gridspec.GridSpec` can be passed to
+# :func:`~matplotlib.pyplot.subplots` via the ``gridspec_kw`` parameter.
+# This example recreates the previous figure without directly using a
+# gridspec instance.
+
+gs_kw = dict(width_ratios=widths, height_ratios=heights)
+fig5, f5_axes = plt.subplots(ncols=3, nrows=3, gridspec_kw=gs_kw)
+for r, row in enumerate(f5_axes):
+    for c, ax in enumerate(row):
+        label = 'Width: {}\nHeight: {}'.format(widths[c], heights[r])
+        ax.annotate(label, (0.1, 0.5), xycoords='axes fraction', va='center')
+
+fig5.tight_layout()
 
+
+###############################################################################
+# Using subplot2grid
+# ==================
+#
 # To use :func:`~matplotlib.pyplot.subplot2grid`, you provide geometry of
```

Comment:
maybe one conceptual-level sentence about what `subplot2grid` is relative to gridspec, and why it exists

Reply:
As someone who doesn't think subplot2grid should exist, I'd advocate moving this section down or removing it altogether.  `subplot2grid` uses `gridspec`, but only in a somewhat peverted way, using a new `gridspec` for each call.  That loses any of the logical layout gridspec gives you, so its a strange thing to highlight early in the gridspec tutorial.